dask/dataframe/io/demo.py

```diff
@@ -98,22 +98,38 @@ def __call__(self, part):
         if isinstance(state_data, int):
             state_data = random_state_data(1, state_data)
         return make_timeseries_part(
-            divisions[0], divisions[1], self.dtypes, self.freq, state_data, self.kwargs
+            divisions[0],
+            divisions[1],
+            self.dtypes,
+            self.columns,
+            self.freq,
+            state_data,
+            self.kwargs,
         )
 
 
-def make_timeseries_part(start, end, dtypes, freq, state_data, kwargs):
+def make_timeseries_part(start, end, dtypes, columns, freq, state_data, kwargs):
     index = pd.date_range(start=start, end=end, freq=freq, name="timestamp")
     state = np.random.RandomState(state_data)
-    columns = {}
+    data = {}
     for k, dt in dtypes.items():
         kws = {
             kk.rsplit("_", 1)[1]: v
             for kk, v in kwargs.items()
             if kk.rsplit("_", 1)[0] == k
         }
-        columns[k] = make[dt](len(index), state, **kws)
-    df = pd.DataFrame(columns, index=index, columns=sorted(columns))
+        # Note: we compute data for all dtypes in order, not just those in the output
+        # columns. This ensures the same output given the same state_data, regardless
+        # of whether there is any column projection. So there is extra work done for
+        # the demo dataset in the presence of projection.
+        # It might be nice to use the internal `advance(n_samples)` method on the random
+        # state to avoid that extra work, but that is probably unreliable as different
+        # internal distributions reserve the right to advance the state as much as
+        # they want (e.g., for rejection sampling).
```

Comment:
There is a lot of info in this comment, but the information is certainly important. Any interest in making this comment brief, but referencing a more-detailed explaination in this PR?

Reply:
Sure, I can do that